repos/system_upgrade/el8toel9/actors/checkvdo/libraries/checkvdo.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from leapp import exceptions
+from leapp import models
+from leapp import reporting
+from leapp.libraries import stdlib
+
+def _run_cmd(command, checked = True):
+    """ Run the specified command returning the result. """
+    try:
+        result = stdlib.run(command, checked = checked)
+    except stdlib.CalledProcessError:
+        raise exceptions.StopActorExecutionError('Failure to execute "{0}"'.format(' '.join(command)))
+    return result
+
+def _canonicalize_device_path(path):
+    return os.path.realpath(path)
+
+#
+# Methods used in discovering VDO devices which require migration to
+# lvm-based management.
+#
+def _get_unmigrated_vdo_blkid_results():
+    """ Collect devices that have not been migrated to lvm-based management. """
+    # VDO devices that have been created by or migrated to lvm-based management
+    # are not identified by blkid.
+    command = ['blkid', '--output', 'device', '--match-token', 'TYPE=vdo']
```

Comment:
@jshimkus-rh  we prohibit interaction with the system in the `ChecksPhase` phase. Imagine that actors in checksphase should be written in way you can execute them on a different system, so you have only data from the scanner phase.

Reply:
also, it seems to me like an info that could be included possibly in one of storageinfo msgs (see storageinfo.py in models). See the storagescanner actor in the common repository. Wdyt about an extention of that actor?

Reply:
For clarity, generating inhibitory reports during `FactsPhase` is considered acceptable?  Or should a `FactsPhase`  produce the necessary device info to be processed in a `ChecksPhase` which would generate the inhibitory report?

Reply:
@pirat89 
I'm still looking for answers to my questions of a week ago as they relate to the amount of additional work that would be required.  Switching to `FactsPhase` is obviously trivial; if that's not acceptable generating new fact output, consuming same would be require more effort (whether storage scanner extension or not).
I'm not wedded to either, just looking for direction.

Reply:
@jshimkus-rh I am sorry, I had really terrible (super super busy) start of the year and missed this completely. Only in some exceptions we close our eyes to generate report during `FactsPhase`. Usually we want to keep scanners to be just scanners, and keep the reporting on actors in the checksphase. But there are exceptions when it makes sense to put the report into the `FactsPhase` phase.